### PR TITLE
[USM] Add raw test for CONTINUATION frame

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?

Adds a raw test for the `CONTINUATION` frame.

### Motivation

This pull request adds a test for `CONTINUATION` frames, validating the lack of support for `CONTINUATION` frames.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 1]
```diff
@@ -630,7 +630,7 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 				for i := 0; i < iterations; i++ {
 					streamID := getStreamID(i)
 					framer.
-						writeRawHeaders(t, streamID, headersFrame).
+						writeRawHeaders(t, streamID, true, headersFrame).
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:633]: maybe define
```go
endHeaders = true
```

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 2]
```diff
@@ -1103,6 +1103,36 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 				}: 2,
 			},
 		},
+		{
+			name: "validate CONTINUATION frame support",
+			// Testing the scenario in which part of the data frame header is sent using CONTINUATION frame.
+			// Currently, we do not support CONTINUATION frames, therefore, we expect to capture only the first
+			// part of the message.
+			messageBuilder: func() [][]byte {
+				preFixTestHeaders, sufFixTestHeaders := splitTestHeaders(2)
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:1112]:
```suggestion
				prefixTestHeaders, suffixTestHeaders := splitTestHeaders(2)
```

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 3]
```diff
@@ -1384,6 +1414,11 @@ func headersWithGivenEndpoint(path string) []hpack.HeaderField {
 // testHeaders returns a set of header fields.
 func testHeaders() []hpack.HeaderField { return generateTestHeaderFields(headersGenerationOptions{}) }
 
+// splitTestHeaders returns a set of header fields spited by the given index.
+func splitTestHeaders(index int) ([]hpack.HeaderField, []hpack.HeaderField) {
+	return generateTestHeaderFields(headersGenerationOptions{})[:index], generateTestHeaderFields(headersGenerationOptions{})[index:]
+}
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:1420]: I don't see a value for this function, just put it inline

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 4]
```diff
@@ -1103,6 +1103,36 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
...
+				require.NoError(t, err, "could not create headers frame")
+
+				sufFixHeadersFrame, err := usmhttp2.NewHeadersFrameMessage(usmhttp2.HeadersFrameOptions{
+					Headers: sufFixTestHeaders,
+				})
+				require.NoError(t, err, "could not create headers frame")
+
+				headersFrame := newFramer().writeRawHeaders(t, 1, false, prFixHeadersFrame).bytes()
+				continuationFrame := newFramer().writeRawContinuation(t, 1, true, sufFixHeadersFrame).bytes()
+				dataFrameSecond := newFramer().writeData(t, 1, true, emptyBody).bytes()
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:1125]: you should be using the same framers, as the indexing between the headers and the continuation should match

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 5]
```diff
@@ -1103,6 +1103,36 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
...
+
+				firstMessage := headersFrame
+				secondMessage := append(continuationFrame, dataFrameSecond...)
+				return [][]byte{
+					firstMessage,
+					secondMessage,
+				}
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:1132]: what do you benefit from splitting the message into 2 different TCP writes?

**Review by guyarb — CHANGES_REQUESTED** [review]
Reviewed

File: pkg/network/usm/usm_http2_monitor_test.go (hunk 1 above)
Comment (amitslavin) [reply pkg/network/usm/usm_http2_monitor_test.go:633]: added to the whole file

File: pkg/network/usm/usm_http2_monitor_test.go (hunk 2 above)
Comment (amitslavin) [reply pkg/network/usm/usm_http2_monitor_test.go:1112]: removed

File: pkg/network/usm/usm_http2_monitor_test.go (hunk 3 above)
Comment (amitslavin) [reply pkg/network/usm/usm_http2_monitor_test.go:1420]: removed

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 6]
```diff
@@ -506,12 +506,13 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 			// a single program.
 			messageBuilder: func() [][]byte {
 				const settingsFramesCount = 238
+				const endStream = true
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:509]: make it a global const, no much point to have it defined multiple times

Comment (guyarb) [reply pkg/network/usm/usm_http2_monitor_test.go:509]: I'm confused
The original intention was to control endHeaders, but the change in the file is only for end stream

**Review by guyarb — CHANGES_REQUESTED** [review]

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 7]
```diff
@@ -1103,6 +1126,36 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
...
 		},
+		{
+			name: "validate CONTINUATION frame support",
+			// Testing the scenario in which part of the data frame header is sent using CONTINUATION frame.
+			// Currently, we do not support CONTINUATION frames, therefore, we expect to capture only the first
+			// part of the message.
+			messageBuilder: func() [][]byte {
+				const endStream, continuationFrameEndHeaders, headersFrameEndHeaders = true, true, false
+				prefixHeadersFrame, err := usmhttp2.NewHeadersFrameMessage(usmhttp2.HeadersFrameOptions{
+					Headers: generateTestHeaderFields(headersGenerationOptions{})[:2],
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:1137]: call `generateTestHeaderFields(headersGenerationOptions{})` only once

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 8]
```diff
@@ -1103,6 +1126,36 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
...
+				require.NoError(t, err, "could not create prefix headers frame")
+
+				suffixHeadersFrame, err := usmhttp2.NewHeadersFrameMessage(usmhttp2.HeadersFrameOptions{
+					Headers: generateTestHeaderFields(headersGenerationOptions{})[2:],
+				})
+				require.NoError(t, err, "could not create suffix headers frame")
+
+				headersFrame := newFramer().writeRawHeaders(t, 1, headersFrameEndHeaders, prefixHeadersFrame).bytes()
+				continuationFrame := newFramer().writeRawContinuation(t, 1, continuationFrameEndHeaders, suffixHeadersFrame).bytes()
+				dataFrameSecond := newFramer().writeData(t, 1, endStream, emptyBody).bytes()
```
Comment (guyarb) [inline pkg/network/usm/usm_http2_monitor_test.go:1148]: convert to a single write

**Review by guyarb — CHANGES_REQUESTED** [review]

**Review by guyarb — APPROVED** [review]

**Pull request merged**